src/main.rs

```diff
@@ -118,6 +121,33 @@ fn run() -> i32 {
         },
     };
 
+    if matches.opt_present("s") {
+        let cursor = match display.get_cursor_position() {
+            Some(p) => p,
+            None => {
+                eprintln!("Failed to get cursor position");
+                return 1;
+            }
+        };
+
+        let screen_rects = match display.get_screen_rects(display.get_default_root()) {
+            Some(r) => r,
+            None => {
+                eprintln!("Failed to get screen rects");
+                return 1;
+            }
+        };
+
+        // Find the screen that the cursor is on
+        sel = match screen_rects.enumerate().find(|(_, r)| r.contains(cursor)) {
+            Some((_, r)) => r,
+            None => {
+                eprintln!("Failed to find screen containing cursor");
+                return 1;
+            }
+        }
```

Comment:
Alright so under normal circumstances this shouldn't fail. That said, maybe we should only bother to mutate `sel` if something is found? Honestly not sure what the best strategy is here. This seems sound enough all things considered.